internal/apiresource/pipeline.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright IBM Corporation 2020
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiresource
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/konveyor/move2kube/internal/common"
+	irtypes "github.com/konveyor/move2kube/internal/types"
+	"github.com/konveyor/move2kube/internal/types/tekton"
+	plantypes "github.com/konveyor/move2kube/types/plan"
+	log "github.com/sirupsen/logrus"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	pipelineKind              = "Pipeline"
+	defaultGitRepoBranch      = "master"
+	gitRepoURLPlaceholder     = "<TODO: insert git repo url>"
+	contextPathPlaceholder    = "<TODO: insert path to the directory containing Dockerfile>"
+	dockerfilePathPlaceholder = "<TODO: insert path to the Dockerfile>"
+)
+
+// Pipeline handles all objects like a Tekton pipeline.
+type Pipeline struct {
+}
+
+// GetSupportedKinds returns the kinds that this type supports.
+func (*Pipeline) GetSupportedKinds() []string {
+	return []string{pipelineKind}
+}
+
+// CreateNewResources creates the runtime objects from the intermediate representation.
+func (p *Pipeline) CreateNewResources(ir irtypes.IR, supportedKinds []string) []runtime.Object {
+	irresources := ir.TektonResources.Pipelines
+	objs := []runtime.Object{}
+	for _, irresource := range irresources {
+		objs = append(objs, p.createNewResource(irresource, ir))
+	}
+	return objs
```

Comment:
we should check if it is a supported kind.

Reply:
done